packages/common/src/tools/queue/TaskQueue.ts

```diff
@@ -153,3 +160,15 @@ export class TaskQueue<T> {
     }
   }
 }
+
+function wrapAndMeasure<T extends unknown[], U>(
```

Comment:
i wonder can this be shared across this and BaseRepository's wrap

Reply:
That was probably my goal looking at the generic types here. Will do